hls4ml/model/attributes.py

```diff
@@ -79,24 +80,23 @@ class TypeAttribute(Attribute):
     As a convention, the name of the attribute storing a type will end in ``_t``.
     """
 
-    def __init__(self, name, default=None, configurable=True):
+    def __init__(self, name, default=None, configurable=True, description=None):
         if not name.endswith('_t'):
             name += '_t'
-        super().__init__(name, value_type=NamedType, default=default, configurable=configurable)
+        super().__init__(name, value_type=NamedType, default=default, configurable=configurable, description=description)
 
 
 class ChoiceAttribute(Attribute):
     """
     Represents an attribute whose value can be one of several predefined values.
     """
 
-    def __init__(self, name, choices, default=None, configurable=True):
-        super().__init__(name, value_type=list, default=default, configurable=configurable)
+    def __init__(self, name, choices, default=None, configurable=True, description=None):
+        super().__init__(name, value_type=list, default=default, configurable=configurable, description=description)
         assert len(choices) > 0
         if default is not None:
             assert default in choices
         self.choices = choices
-        self.value_type = str(self.choices)
```

Comment:
I just noticed this change. In the  base class, there is no casting to string. It seems a bit unrelated to the PR, but I wanted to understand why it was this was originally and why it changed.

Reply:
Not really sure what the original idea was, it seems like a mistake that has gone unnoticed. I don't see why we would be doing that, and it breaks the script, so I removed it